core/src/main/scala/org/apache/spark/scheduler/TaskScheduler.scala

```diff
@@ -109,4 +109,13 @@ private[spark] trait TaskScheduler {
    */
   def applicationAttemptId(): Option[String]
 
+  /**
+   * SPARK-25250: Whenever any Task gets successfully completed, we simply mark the
+   * corresponding partition id as completed in all attempts for that particular stage and
+   * additionally, for a Result Stage, we also kill the remaining task attempts running on the
+   * same partition. As a result, we do not see any Killed tasks due to
+   * TaskCommitDenied Exceptions showing up in the UI.
```

Comment:
I wouldn't mention the TaskCommitDenied bit here -- that's really just one very particular thing which can go wrong later.

Also I think its worth mentioning that this method must be called from inside the DAGScheduler event loop, to ensure a consistent view of all task sets for the given stage.

Reply:
Done